keps/sig-node/20181106-in-place-update-of-pod-resources.md

```diff
@@ -200,6 +227,34 @@ Spec.Containers[i].ResourcesAllocated are used to determine if there is enough
 room to admit the Pod. Kubelet does not set Pod's ResourcesAllocated after
 admitting a new Pod.
 
+<<[UNRESOLVED]>>
+The above is confusing.  Does it mean that ResourcesAllocated will be set by
+someone other than Kubelet?  Who?  Why?
```

Comment:
Pod creator/user is the other entity who can optionally specify ResourcesAllocated field in podSpec - only at the time of creation of new pod. They cannot update it (the new admission control plugin prevents it, and errors out if user sets it to anything other than Resources.Requests for now.

This came up during a discussion that user could leverage the new field to create a new pod saying "I'd like X but can make-do with Y (Y < X)". It is not in the scope of this KEP. I think there's a comment somewhere, I'll update it if I find the discussion.

Reply:
I find the mental-model to be confusing.  Why can't we say:

- users can not set `allocated` - it must be unset at create
- only kubelet should set `allocated`, period.

Reply:
This is true today - and we enforce ```either don't set or set allocated equal to resources```.
I think we worded it this way to leave the door open for above scenario in the future, where pod can say "on create X is my ideal, Y is minimum"

Reply:
OK.  The wording here is what threw me, then.  I am a simple man and I like simple sentences.   :P